exchanges/gateio/gateio_websocket.go

```diff
@@ -622,19 +635,24 @@ func (g *Gateio) processCrossMarginBalance(data []byte) error {
 	if err != nil {
 		return err
 	}
-	accountChanges := make([]account.Change, len(resp.Result))
+	creds, err := g.GetCredentials(context.Background())
+	if err != nil {
+		return err
+	}
+	changes := make([]account.Change, len(resp.Result))
 	for x := range resp.Result {
-		code := currency.NewCode(resp.Result[x].Currency)
-		accountChanges[x] = account.Change{
-			Exchange: g.Name,
-			Currency: code,
-			Asset:    asset.Margin,
-			Amount:   resp.Result[x].Available.Float64(),
-			Account:  resp.Result[x].User,
+		changes[x] = account.Change{
+			Account:   resp.Result[x].User,
+			AssetType: asset.Margin,
+			Balance: &account.Balance{
+				Currency:  currency.NewCode(resp.Result[x].Currency),
+				Total:     resp.Result[x].Available.Float64(),
```

Comment:
```suggestion
				Total:     resp.Result[x].Total.Float64(),
```
Unless the struct is invalid and needs to be updated, in which case, remove the `Total` field

Looks like you could also get freeze: https://www.gate.io/docs/developers/apiv4/ws/en/#client-subscription-15